test/testunusedvar.cpp

```diff
@@ -3239,6 +3240,26 @@ class TestUnusedVar : public TestFixture {
         ASSERT_EQUALS("", errout.str());
     }
 
+    void localvar61() { // #9407
+        functionVariableUsage("void g(int& i);\n"
+                              "void f() {\n"
+                              "    int var = 0;\n"
+                              "    g(var);\n"
+                              "    var = 2;\n"
```

Comment:
how about this?
```
void a();
void b(int &);
void c() {
    int x;
    b(x);
    x = 34;
    a();
}
```

Reply:
I get a warning. There is no other variable than `x`.

Reply:
In theory .. function `b` could take the value of `x` and then in `a` it could be read.

We could test how it works in practice but in theory false positives are possible.

Reply:
In theory, yes. Although I can't think of a use case for storing a reference to a local variable outside of its scope. The second test case already triggers a cppcheck warning for that.

Reply:
From that point of view, the check for variables in the outer scope might as well be removed, since there is no guarantee that global variables are visible anyway.

Reply:
yes. if daca@home will show that there will in practice be noise then we need to adjust. if most of the related daca@home results are true positives we can keep this heuristic. I only say there can be false positives I can't say how frequent it is. So I suggest that we run test-my-pr on this.

Reply:
> Although I can't think of a use case for storing a reference to a local variable outside of its scope. 

to clarify the reference can't be stored I fear it will store the address.

For instance the standard function strtok works more or less like that.

Maybe when you pass a reference it's very uncommon that the address is stored. I don't know.

Reply:
ValueFlow doesn't make such assumptions about code, and there doesn't seem to be too much FPs from that.